src/op_analytics/datapipeline/etl/blockbatchloaddaily/ddl/blockbatch_daily/wallet_segments/daily_tx_activity_segments_v1__INSERT.sql

```diff
@@ -0,0 +1,175 @@
+  avg(sum_l2_priority_fee_native) AS avg_l2_priority_fee_native_per_wallet,
+  avg(sum_success_l2_priority_fee_native) AS avg_success_l2_priority_fee_native_per_wallet,
+  avg(sum_l2_legacy_extra_fee_native) AS avg_l2_legacy_extra_fee_native_per_wallet,
+  avg(sum_success_l2_legacy_extra_fee_native) AS avg_success_l2_legacy_extra_fee_native_per_wallet,
+  avg(sum_l1_fee_native) AS avg_l1_fee_native_per_wallet,
+  avg(sum_success_l1_fee_native) AS avg_success_l1_fee_native_per_wallet,
+  avg(sum_l1_base_fee_native) AS avg_l1_base_fee_native_per_wallet,
+  avg(sum_success_l1_base_fee_native) AS avg_success_l1_base_fee_native_per_wallet,
+  avg(sum_l1_blob_fee_native) AS avg_l1_blob_fee_native_per_wallet,
+  avg(sum_success_l1_blob_fee_native) AS avg_success_l1_blob_fee_native_per_wallet,
+  -- Gas Prices
+  avg(avg_l2_gas_price_gwei) AS avg_l2_gas_price_gwei_across_wallets,
+  avg(avg_l2_priority_gas_price_gwei) AS avg_l2_priority_gas_price_gwei_across_wallets,
+  avg(avg_l2_legacy_extra_gas_price_gwei) AS avg_l2_legacy_extra_gas_price_gwei_across_wallets,
+  avg(avg_l1_base_gas_price_gwei) AS avg_l1_base_gas_price_gwei_across_wallets,
+  avg(avg_l1_blob_base_gas_price_gwei) AS avg_l1_blob_base_gas_price_gwei_across_wallets,
+  -- Input Data
+  avg(sum_input_zero_bytes) AS avg_input_zero_bytes_per_wallet,
+  avg(sum_success_input_zero_bytes) AS avg_success_input_zero_bytes_per_wallet,
+  avg(sum_input_bytes_length) AS avg_input_bytes_length_per_wallet,
+  avg(sum_success_input_bytes_length) AS avg_success_input_bytes_length_per_wallet,
+  avg(sum_input_nonzero_bytes) AS avg_input_nonzero_bytes_per_wallet,
+  avg(sum_success_input_nonzero_bytes) AS avg_success_input_nonzero_bytes_per_wallet,
+  -- Estimated Size
+  avg(sum_estimated_size) AS avg_estimated_size_per_wallet,
+  avg(sum_success_estimated_size) AS avg_success_estimated_size_per_wallet,
+
+  -- === AVERAGES PER TRANSACTION ===
+  avgWeighted(count_success_transactions, count_transactions) AS overall_success_rate,
+  avgWeighted(sum_tx_fee_native, count_transactions) AS overall_avg_tx_fee_native_per_tx,
+  avgWeighted(sum_l2_gas_used, count_transactions) AS overall_avg_l2_gas_per_tx,
+  avgWeighted(sum_l1_gas_used_unified, count_transactions) AS overall_avg_l1_gas_per_tx,
+
+  -- === AVERAGES PER WALLET ===
+  avg(wallet_success_rate) AS avg_wallet_success_rate,
+  avgWeighted(wallet_avg_fee_per_tx, count_transactions) AS avg_wallet_avg_fee_per_tx,
+  avgWeighted(wallet_avg_l2_gas_per_tx, count_transactions) AS avg_wallet_avg_l2_gas_per_tx,
+  avgWeighted(wallet_avg_l1_gas_per_tx, count_transactions) AS avg_wallet_avg_l1_gas_per_tx
+
+FROM DailyWalletCohorts
+-- Group by date, chain, network, and segment, AND also group by just date, chain, network (for baseline)
+GROUP BY GROUPING SETS (
+  (dt, chain, chain_id, network, segment),
+  (dt, chain, chain_id, network)
+)
+-- Order results for readability
+ORDER BY
```

Comment:
Is this is unnecessary in production?

Also wonder if including the `multiIf` logic here as well will be hard to maintain vs 1,2,3,4

And should it be the `wallet_segment` field instead?